agent/config.go

```diff
@@ -22,6 +22,9 @@ type Config struct {
 	// Conn specifies the client connection Agent will use
 	Conn *grpc.ClientConn
 
+	// Picker is the picker used by Conn
+	Picker *picker.Picker
```

Comment:
Could you place a TODO indicating that this is just here to reset the connection?

I actually did some refactoring work to get this out of here, as it presents some awful coupling issues.